proxy_test.go

```diff
@@ -289,3 +290,31 @@ func testConnectResponseHeaders(t *testing.T, server string, client net.Conn) {
 	assert.Empty(t, resp.TransferEncoding)
 	assert.Equal(t, int64(-1), resp.ContentLength)
 }
+
+func TestConnectResponseHasCorrectNewlines(t *testing.T) {
```

Comment:
This test doesn't really belong in this code base because it is not responsible now for the correct formatting of responses. But obviously you want the test because it had the bug. So perhaps a comment on why this test is here might be warranted. Up to you.

Reply:
Yeah I was on the fence about adding a test, and thought it might be good to have one in case anyone decides that building our own `http.Response` and writing it out is overkill compared to just writing a hard-coded string. I've changed the comment slightly, hope it's clear enough...

Reply:
I actually completely missed the link to the issue. I just saw the rfc link and thought it was all the same. Better now